lighthouse/__init__.py

```diff
@@ -27,28 +25,35 @@ def create_app() -> Eve:
     # setup logging
     logging.config.dictConfig(app.config["LOGGING"])
 
-    from lighthouse.blueprints import (
-        beckman,
-        cherrypicked_plates,
-        plate_events,
-        plates,
-        reports,
-    )
-
-    app.register_blueprint(plates.bp)
-    app.register_blueprint(reports.bp)
-
-    if app.config.get("BECKMAN_ENABLE", False):
-        app.register_blueprint(beckman.bp)
-        app.register_blueprint(cherrypicked_plates.bp)
-        app.register_blueprint(plate_events.bp)
-
     if app.config.get("SCHEDULER_RUN", False):
         scheduler.init_app(app)
         scheduler.start()
 
+    setup_routes(app)
+
     @app.get("/health")
-    def health_check():
+    def _():
+        """Confirms the health of Lighthouse by confirming it is responding to requests."""
         return "Factory working", HTTPStatus.OK
 
     return app
+
+
+def setup_routes(app):
+    # Register the v1 endpoints for the Eve API.  Note that Eve automatically registers endpoints at the root.
+    from lighthouse.routes import eve_routes
+
+    app.register_blueprint(eve_routes.bp, url_prefix="/v1")
+
+    # When registering blueprints, do so both in the root and in /v1.
+    # Future versions will just be appended to the bottom of these registrations.
+    from lighthouse.routes.v1 import routes as v1_routes
+
+    app.register_blueprint(v1_routes.bp)
+    app.register_blueprint(v1_routes.bp, url_prefix="/v1")
+
+    if app.config.get("BECKMAN_ENABLE", False):
+        from lighthouse.routes.v1 import beckman_routes as v1_beckman_routes
+
+        app.register_blueprint(v1_beckman_routes.bp)
+        app.register_blueprint(v1_beckman_routes.bp, url_prefix="/v1")
```

Comment:
Why serve these routes if no one will consume them?

Reply:
I don't think we want to introduce versioning and then make each versions be only a partial API.  That would get unmanageable quite quickly if users can't just rely on the base path staying consistent between different endpoints.  Maybe no-one will use these endpoints, but it doesn't matter.  I would still expect them to get served on `v2` in the future when that version is released, as well.

Also, my aim here was to get every endpoint that was previously hosted at the root, to be part of `v1`.  Ideally, after a long time, you could decommission the root API as everyone will have moved up to the latest version.  If you don't even serve on the latest version, they can never move up.

Reply:
I think I understand your reasoning now and it makes more sense than what I was suggesting, thanks!